src/include/duckdb/storage/single_file_block_manager.hpp

```diff
@@ -49,10 +107,17 @@ class SingleFileBlockManager : public BlockManager {
 
 	FileOpenFlags GetFileFlags(bool create_new) const;
 	//! Creates a new database.
-	void CreateNewDatabase();
+	void CreateNewDatabase(string *encryption_key = nullptr);
```

Comment:
nit: we're trying to get rid of raw C-style pointers everywhere - can this be `optional_ptr<string>` instead?